doc/src/manual/types.md

```diff
@@ -1127,6 +1058,123 @@ must always be specified in full, this is not especially helpful, but in Julia,
 to write just `Vector` for the abstract type including all one-dimensional dense arrays of any
 element type.
 
+## [Singleton types](@id man-singleton-types)
+
+Immutable composite types with no fields are called *singletons*. Formally, if 
+`a isa T && b isa T` implies `a === b`, then `T` is a singleton type. [^2]
```

Comment:
I think the spirit of this formalism is great, but if I have

```julia
julia> abstract type A end

julia> struct B <: A end
```

and I don't define any other subtype of `A`, would it imply that `A` is also a singleton type?

Maybe making this perfect requires `isconcretetype(T)`?  Although if you say "for any `a` and `b`" I guess it could mean "for any `a` and `b` of any implementable type."

> should I mention that abstract types can't be singleton types, or is it obvious from the context?

So yes, I think this helps to make this definition clearer.

Reply:
Yes, I think we should stress that only concrete types can to be singletons.